config/deploy/01-metrics-server-tls-secret.yaml

```diff
@@ -0,0 +1,14 @@
+# This Secret is only for testing / dev.
+# This cert is valid till May 21 12:11:09 3021 GMT
```

Comment:
dont expire !

Reply:
I tried looking into that after your initial suggestion but found that apparently, you can't generate non-expiring certs via openssl. People usually tried hacky ways like creating certs expiring in the year 9999 and what not.

Ref: https://security.stackexchange.com/questions/49824/is-there-such-a-thing-as-a-non-expiring-ssl-certificate

Another ref where the conversation became quite hostile xD : https://www.linuxquestions.org/questions/linux-server-73/https-create-certificate-that-does-not-expire-519699/

In productionized scenarios, we don't have to worry because of service-ca-operator regenerating the certs on the fly as soon as they reach their EOL. And the good part about kube-auth-proxy is that it has relevant watchers setup to make it cert-change-aware and synchronise itself whenever the certs are updated (in our case, when service-ca-operator would do so) in the configured certDir.
So we're good on that end as well.